runtime/doc/lua.txt

```diff
@@ -1571,6 +1571,75 @@ schedule_wrap({cb})                                      *vim.schedule_wrap()*
       • |vim.schedule()|
       • |vim.in_fast_event()|
 
+system({spec}, {on_exit})                                       *vim.system()*
+    Run a system command
+
+    Examples: >lua
+
+      local on_exit = function(code, signal, stdout, stderr)
+        print(code)
+        print(signal)
+        print(stdout)
+        print(stderr)
+      end
+
+      -- Run asynchronously
+      vim.system({'echo', 'hello'}, on_exit)
+      vim.system({cmd = {'echo', 'hello'}}, on_exit)
+      vim.system({cmd = 'echo', args = {'hello'}}, on_exit)
```

Comment:
Nice shape! It's a good pattern for the callback to be the last arg because it tends to be lexically more noisy than other args.

Reply:
Yes. I believe continuation callbacks (callbacks that are only every called once) should always be positioned as the last arguments (after opts).

Mostly because:
- to be consistent with luv
- to synergize with the future async lib which transforms functions of the form `function(<args>, cb(<ret)>))` -> `function(<args>) -> <ret>`.